src/internal_events/dnstap.rs

```diff
@@ -1,41 +1,52 @@
-// ## skip check-events ##
-
 use metrics::counter;
 use vector_core::internal_event::InternalEvent;
 
 #[derive(Debug)]
-pub struct DnstapEventReceived {
+pub struct DnstapEventsReceived {
     pub byte_size: usize,
 }
 
-impl InternalEvent for DnstapEventReceived {
+impl InternalEvent for DnstapEventsReceived {
     fn emit_logs(&self) {
-        trace!(message = "Received line.", byte_size = %self.byte_size);
+        trace!(message = "Events received.", count = 1, byte_size = %self.byte_size);
     }
 
     fn emit_metrics(&self) {
-        counter!("processed_events_total", 1);
         counter!("component_received_events_total", 1);
+        counter!(
+            "component_received_event_bytes_total",
+            self.byte_size as u64
+        );
+        // deprecated
+        counter!("processed_events_total", 1);
         counter!("events_in_total", 1);
         counter!("processed_bytes_total", self.byte_size as u64);
     }
 }
 
 #[derive(Debug)]
-pub struct DnstapParseDataError<'a> {
+pub struct DnstapParseError<'a> {
     pub error: &'a str,
 }
 
-impl<'a> InternalEvent for DnstapParseDataError<'a> {
+impl<'a> InternalEvent for DnstapParseError<'a> {
     fn emit_logs(&self) {
         error!(
-            target = "dnstap event",
             message = "Error occurred while parsing dnstap data.",
             error = ?self.error,
-            internal_log_rate_secs = 10);
+            error_type = "parse_failed",
+            stage = "processing",
+            internal_log_rate_secs = 10,
+        );
     }
 
     fn emit_metrics(&self) {
+        counter!(
+            "component_errors_total", 1,
+            "stage" => "processing",
+            "error_type" => "parse_failed",
```

Comment:
Missed this on my first pass, but we can add the `error` as a tag here.

Reply:
👍 thanks, I missed it (again)